public/Assets/js/domListeners.js

```diff
@@ -112,7 +112,54 @@ document.getElementById("SLexpandBtn").addEventListener("click", function() {
         SLCont.classList.add("expanded");
         this.classList.add("expanded");
     }
-})
+});
+
+// Search Button
+document.getElementById("searchButton").addEventListener("click", async function() {
+    const query = document.getElementById("searchQuery").value;
+    if(!query || query.length == 0) return;
+    if (!inVoiceChannel) {
+        alert("The bot is not in a voice channel");
+        return;
+    }
+
+    /* todo:
+        if query is already a youtube/spotify/soundcloud URL,
+        we should skip this and just emit the play event for socket
+    */
+    const results = await getSearchResults(query);
+    let resultList = "";
+    for(let i = 0; i < results.length; i++) {
+        resultList = resultList.concat(
+            `<div class="searchResult">
+                <a href="https://youtu.be/${results[i].id}">${results[i].title}</a>
+                <button class="addResult" value="${results[i].id}">+</button>
+             </div>` // results[i].thumbnail.url for the thumbnail image, in case u wanna show it too
+            );
+    }
+    // set results limit to 10 instead of 5?
+    // cos 5 seems too low for the large SLResults div
+    document.getElementById("SLResults").innerHTML = resultList;
+    document.getElementById("searchQuery").value = "";
+
+    document.querySelectorAll(".addResult").forEach(b => b.addEventListener('click', async function () {
```

Comment:
If we add the listeners when the script loads, it won't work because the `.addResult` button initially doesn't exist and only shows up in the HTML structure when search results are retrieved.

A better way do to this would probably be to just add 5 searchResult & addResult elements in the HTML and hide them by default, then only show them when the user does a youtube search and change the element values accordingly - but I was too lazy for that yesterday (and I'm bad at HTML/CSS), so I just set it to add the .addResult button listeners on every search request, which is probably a bad idea.

Reply:
I stared at this for several minutes and I don't really see anything wrong.
when the ``SLResults.innerHTML`` is updated, the buttons are already "existing", so when you do ``document.querySelectorAll(".addResult")`` it should work.

I tested it and it actually worked (logs was shown)
![image](https://user-images.githubusercontent.com/40543984/152650135-b8c5f233-48cb-49c2-9729-4e183f191e07.png)